frontend/src/components/Modal/ConfirmModal/ConfirmModal.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from "react";
+import styles from "./ConfirmModal.module.scss";
+import { confirmUser } from "../../../API/UserService/confirmUser";
+import { toast } from "react-toastify";
+import { useNavigate } from "react-router-dom";
+import { useAppDispatch, useAppSelector } from "store/store";
+import { getProfilesForLoggedUser } from "store/profile";
+import { changeActiveProfile } from "store/activeProfile";
+
+interface ConfirmModalProps {
+    profileID: string | undefined;
+    changeVisiblity: () => void;
+    userID: string;
+    functionToConfirm: () => void;
+    toastName: string;
+}
+
+export const ConfirmModal = ({
+    changeVisiblity,
+    profileID,
+    userID,
+    functionToConfirm,
+    toastName,
+}: ConfirmModalProps) => {
+    const [password, setPassword] = useState("");
+    const [errorMessage, setErrorMessage] = useState(" ");
+    const navigate = useNavigate();
+    const dispatch = useAppDispatch();
+    const userProfiles = useAppSelector((state) => state.profiles.profiles);
+
+    const handleSubmit = async (e: React.FormEvent) => {
+        e.preventDefault();
```

Comment:
you don't really have to pass event if you handle this before this function ie. 
```tsx
onSubmit = {(e)=>{e.preventDefault; handleSubmit()}}
```
this will make handleSubmit fn clearer